pkg/virt-handler/virtwrap/manager.go

```diff
@@ -328,21 +327,14 @@ func VMNamespaceKeyFunc(vm *v1.VM) string {
 	return domName
 }
 
-func (l *LibvirtDomainManager) SyncVM(vm *v1.VM) (*api.DomainSpec, error) {
-	var wantedSpec api.DomainSpec
-	mappingErrs := model.Copy(&wantedSpec, vm.Spec.Domain)
-
-	if len(mappingErrs) > 0 {
-		return nil, errors.NewAggregate(mappingErrs)
-	}
-
+func (l *LibvirtDomainManager) SyncVM(vm *v1.VM, domDesign *designer.DomainDesign) (*api.DomainSpec, error) {
```

Comment:
Ultimately there will be some kind of  "Network" or "NetworkPort" resource exposed, and the VM spec will reference this as the backend for a virtual NIC - likely the public Domain Spec would just contain the name of the resource. virt-handler will resolve this and determine that it needs to be turned into a tap or macvtap or vhostuser or sriov assigned device for the VM XML. When we do that resolution, we need to directly write to the libvirt domain XML - there's nothing in the public Domain Spec data structure that lets us storage info about the tap/macvtap/vhostuser backend config data. Resolving k8s objects & generating libvirt XML has to be single integrated process, not separate steps.